.github/workflows/build_and_test.yaml

```diff
@@ -44,3 +44,14 @@ jobs:
         tools/ci/run_ci.sh --run-tests --sanitizer ${{ matrix.sanitizer }}
       env:
         CC:  ${{ matrix.compiler }}
+
+    - name: Install dependencies
+      run: |
+          python -m pip install --upgrade pip
```

Comment:
Why is this needed here, but (apparently) not for gitlint?

Reply:
It is better to be consistent, I'll remove the pip upgrade.
Do you think we shall "lock down" all versions of everything we use e.g. python requirements? It makes sense to have fixed versions of everything in a CI flow (same environment for every run). On the other hand this is a small project and it might be less of a hassle to fix version issues when/if they arise than to remember to upgrade versions. We probably don't want the complexity of CI "progressions" testing and updating the versions automatically...

Reply:
I am not certain, but I'd leave it as it is. Getting to know if newer versions break our builds has some value too.